---
title: 'Résumé'
date: 22 Aug 2022
draft: false
---

Experience
----------

### Senior Staff Data Engineer

#### Pendulum Therapeutics 2019 — Present
* Worked cross functionally with relevant stakeholders at Pendulum to architect the full stack of software and data infrastructure.
* Architected Python libraries to enable data scientists to easily interact with REST APIs and databases.
* Managed junior software engineers, mentored their technical development with pair programming and code reviews while helping to balance their personal career objectives with company goals.
* Led development of internal web applications. Creating the data model, API, and frontend layer.
* Developed ETL and ELT pipelines for integrating Marketing, Manufacturing, and R&D data from SaaS applications and internal databases into a Snowflake data warehouse using Luigi, Prefect, and DBT.
### Bioinformatics Cloud Engineer

#### Second Genome 2018 — 2019

* Converted bioinformatics pipelines for processing genomic and metagenomic data into serverless, containerized applications using Docker, AWS Batch, AWS Lambda, AWS SQS.
* Spearheaded python code consolidation within the company into a single unified library, enforcing test-driven development and software development lifecycle practices.
* Developed ETL pipelines to automatically scrape NCBI and IMG databases and integrate them with private datasets into data warehouses

### Postdoctoral Research Fellow

#### Caltech 2014―2018

My role was the head bioinformatician for a diverse group of microbiologists, chemists and data scientists at the California Institute of Technology (Caltech). My day to day tasks included the processing of genomic and proteomic data, and generating analysis summaries and plots using statistical software and programing tools such as R and python. As part of this role I was responsible for the production of automated pipelines for data analysis and have developed numerous tools and scripts for processing genomic data. I implemented SQL databases to store the extensive genomic and proteomic data that were generated and developed simple interfaces such that other members of the group could access this information without in-depth knowledge of SQL.

### Graduate Student

#### University of Quensland 2010―2014

My PhD research focused on metagenomic analysis of microbial and viral communities from engineered ecosystems found in wastewater treatment plants. Throughout my PhD I was at the interface of microbial ecology, chemical engineering, and computational biology and have gained many skills ranging from microscopy, reactor operation and computer programming. I developed a number of bioinformatic pipelines and substantial computer software in perl, python and C++. My background in molecular biology allowed me to place the inferences of metagenomic data into a broader biological framework and work with other members of the lab that did not have computer skills.

Education
---------

### University of Queensland

2014, PhD in Microbial Ecology

2010, Bachelor of Biotechnology

### Honors & Awards

2015, Dean's award for outstanding PhD dissertation, University of Queensland

2010, Australian Research Council PhD Fellowship

Publications
------------

McMurdie, P. J., Stoeva, M. K., Justice, N., Nemchek, M., Sieber, C. M. K., Tyagi, S., Gines J., **Skennerton, C. T.,** Souza, M., Kolterman, O., Eid, J. (2022). "[Increased circulating butyrate and ursodeoxycholate during probiotic intervention in humans with type 2 diabetes](https://doi.org/10.1186/s12866-021-02415-8)". BMC Microbiol 22, 19 (2022).

Chadwick, G. L., **Skennerton, C. T.,** Laso-Pérez, R., Leu, A. O., Speth, D. R., Yu, H., Morgan-Lang, C., Hatzenpichler, R., Goudeau, D., Malmstrom, R., Brazelton, W. J., Woyke, T., Hallam, S. J., Tyson, G. W., Wegener, G., Boetius, A., Orphan, V. J. (2022) "[Comparative genomics reveals electron transfer and syntrophic mechanisms differentiating methanotrophic and methanogenic archaea](https://doi.org/10.1371/journal.pbio.3001508)". PLoS Biol 20(1): e3001508.

Yu, H., **Skennerton, C. T.,** Chadwick, G. L., Leu, A. O., Aoki, M., Tyson, G. W., Orphan, V. J. (2022) "[Sulfate differentially stimulates but is not respired by diverse anaerobic methanotrophic archaea](https://pubmed.ncbi.nlm.nih.gov/34285362/)". ISME J 16(1): 168-177.

Sim, M., **Skennerton, C. T.,** Orphan, V. J. (2021) "[Physiological, genomic, and sulfur isotopic characterization of methanol metabolism by Desulfovibrio carbinolicus](https://pubmed.ncbi.nlm.nih.gov/33444327/)". PLoS ONE. 16(1): e0245069

Yu, H., Susanti, D., McGlynn, S. E., **Skennerton, C. T.,** Chourey, K., Iyer, R., Scheller, S., Tavormina, P. L., Hettich, R. L., Mukhopadhyay, B., Orphan V. J. (2018) "[Comparative Genomics and Proteomic Analysis of Assimilatory Sulfate Reduction Pathways in Anaerobic Methanotrophic Archaea](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6286981/)". Frontiers Microbiol. 9: 2917

**Skennerton, C. T.**, Chourey, K., Iyer, R., Hettich, R. L., Tyson, G. W., Orphan, V. J. (2017) "[Methane-Fueled Syntrophy through Extracellular Electron Transfer: Uncovering the Genomic Traits Conserved within Diverse Bacterial Partners of Anaerobic Methanotrophic Archaea](https://www.ncbi.nlm.nih.gov/pubmed/28765215)". mBio 8(4):e00530-17

**Skennerton, C. T.**, Haroon, M. F., Briegel, A., Shi, J., Jensen, G. J., Tyson, G. W., Orphan, V. J. (2016) "[Phylogenomic analysis of _Candidatus_ ‘Izimaplasma’ species: free-living representatives from a Tenericutes clade found in methane seeps](http://www.ncbi.nlm.nih.gov/pubmed/27058507)". ISMEJ. 10: 2679–2692

Marlow, J. J., **Skennerton, C. T.,** Li, Z., Chourey, K., Hettich, R. L., Pan, C., Orphan, V. J. (2016). "[Proteomic Stable Isotope Probing Reveals Biosynthesis Dynamics of Slow Growing Methane Based Microbial Communities](http://journal.frontiersin.org/article/10.3389/fmicb.2016.00563/full)". Frontiers Microbiol. 7: 563

Barr, J. J., Dutilh, B. E., **Skennerton, C. T.**, Fukushima, T., Hastie, M. L., Gorman, J. J., Tyson, G. W., Bond, P. L. (2016). “[Metagenomic and metaproteomic analyses of Accumulibacter phosphatis enriched floccular and granular biofilm](http://www.ncbi.nlm.nih.gov/pubmed/26279094)”. Environ. Microbiol. 18(1): 273–287

**Skennerton, C. T.,** Ward, L. M., Michel, A., Metcalfe, K., Valiente, C., Mullin, S., Chan, K. Y., Gradinaru, V., Orphan, V. J. (2015). "[Genomic Reconstruction of an Uncultured Hydrothermal Vent Gammaproteobacterial Methanotroph (Family _Methylothermaceae_) Indicates Multiple Adaptations to Oxygen Limitation](http://journal.frontiersin.org/article/10.3389/fmicb.2015.01425/full)". Frontiers Microbiol. 6

Parks, D. H., Imelfort, M., **Skennerton, C. T.**, Hugenholtz, P., Tyson, G. W. (2015). “[CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes](http://genome.cshlp.org/content/25/7/1043.long)”. Genome Res. 25: 1043–1055.

**Skennerton, C. T.**, Barr, J. J., Slater, F. R., Bond, P. L., Tyson, G. W. (2015). “[Expanding our view of genomic diversity in _Candidatus_ Accumulibacter clades](http://www.ncbi.nlm.nih.gov/pubmed/25088527)”. Environ. Microbiol. 17 (5): 1574–1585.

Soo, R. M., **Skennerton, C. T.**, Sekiguchi, Y., Imelfort, M., Paech, S. J., Dennis, P. G., Steen, J. A., Parks, D. H., Tyson, G. W., Hugenholtz, P. (2014). “[An expanded genomic representation of the phylum cyanobacteria](http://gbe.oxfordjournals.org/content/6/5/1031.long)”. Genome Biol. Evol. 6 (5): 1031– 1045.

Haroon, M. F., **Skennerton, C. T.**, Steen, J. A., Lachner, N., Hugenholtz, P., Tyson, G. W. (2013). “[In-Solution Fluorescence In Situ Hybridization and Fluorescence-Activated Cell Sorting for Single Cell and Population Genome Recovery](http://www.ncbi.nlm.nih.gov/pubmed/24060113)”. In: Microbial Metagenomics, Metatranscriptomics, and Metaproteomics. Ed. by Edward F. DeLong. Vol. 531. Academic Press, 3–19.

**Skennerton, C. T.**, Imelfort, M., Tyson, G. W. (2013). “[Crass: identification and reconstruction of CRISPR from unassembled metagenomic data](http://nar.oxfordjournals.org/content/41/10/e105.long)”. Nucleic Acids Res. 41 (10): e105.

**Skennerton, C. T.**, Angly, F. E., Breitbart, M., Bragg, L., He, S., McMahon, K. D., Hugenholtz, P., Tyson, G. W. (2011) “[Phage encoded H-NS: a potential achilles heel in the bacterial defence system](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0020095)”. PLoS One, 6 (5): e20095.
