---
title: "The pace of genome binning from metagenomes"
date: 2015-06-17
tags: ['bioinformatics']
---
With the pace of science what seemed top stuff three years ago is now an
order of magnitude less than what just got published.

<!--more-->
When I was in my PhD I was working on
[EBPR](https://en.wikipedia.org/wiki/Enhanced_biological_phosphorus_removal)
communities using metagenomics to characterize the the microbial and
phage populations. From all of my microbial data I could get about 70 draft
bacterial genomes. At the time (2012) that was pretty huge  considering the first
big paper on genome binning from metagenomes got 49 draft bacteria ([Wrighton et al, 2012](http://www.sciencemag.org/content/337/6102/1661)).
The next year a similar study from EBPR recovered 31 draft genome bins
([Albertsen et al,
2013](http://www.nature.com/nbt/journal/v31/n6/abs/nbt.2579.html)). One of the big results
from all this computational effort is that a number of genomes from
[candidate phyla](https://en.wikipedia.org/wiki/Candidate_division). In
2012 I was super excited that I had multiple genomes from
[Saccharibacteria](https://en.wikipedia.org/wiki/Candidate_division_TM7)
and
[Microgenomates](http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=67820)
which at the time had only a few genome representatives. At the time I
thought I had the richest dataset in the world (which I most definitely
did not) now three years later and ~800 genome bins just got published
mostly from those candidate phyla I just mentioned ([Brown et al, 2015](http://www.nature.com/nature/journal/vaop/ncurrent/full/nature14486.html)). 
It seems like the magnitude of the cutting edge just went up a notch...
